src/codegen/mod.rs

```diff
@@ -3871,4 +3952,48 @@ mod utils {
 
         args
     }
+
+    pub fn fnsig_block(
+        ctx: &BindgenContext,
+        sig: &FunctionSig,
+    ) -> quote::Tokens {
+        use super::ToPtr;
+
+        let args = sig.argument_types().iter().map(|&(_, ty)| {
+            let arg_item = ctx.resolve_item(ty);
+            let arg_ty = arg_item.kind().expect_type();
+
+            match *arg_ty.canonical_type(ctx).kind() {
+                TypeKind::Array(t, _) => {
+                    t.to_rust_ty_or_opaque(ctx, &())
+                        .to_ptr(ctx.resolve_type(t).is_const())
+                },
+                TypeKind::Pointer(inner) => {
+                    let inner = ctx.resolve_item(inner);
+                    let inner_ty = inner.expect_type();
+                    if let TypeKind::ObjCInterface(_) = *inner_ty.canonical_type(ctx).kind() {
```

Comment:
Should this be in Type::to_rust_ty instead, on a pointer branch? Looks like it should to me, then this code would become much simpler. I don't think we really want to restrict this to blocks, right?

Reply:
agree, change to use 
```rust
arg_ty.try_to_rust_ty(ctx, &arg_item)
      .unwrap_or_else(|_| arg_item.to_rust_ty_or_opaque(ctx, &()))
```